packages/contracts/contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol

```diff
@@ -1049,25 +1073,42 @@ contract OVM_CanonicalTransactionChain is iOVM_CanonicalTransactionChain, Lib_Ad
      * Checks that the final batch context in a sequencer submission is valid.
      * @param _finalContext The batch context to validate.
      * @param _queueLength The length of the queue at the start of the batchAppend call.
-     * @param _nextQueueIndex The next element in the queue that will be pulled into the CTC.
+     * @param _firstSubsequentQueueIndex The next element in the queue that will be pulled into the CTC.
      * @param _queueRef The storage container for the queue.
      */
     function _validateFinalBatchContext(
         BatchContext memory _finalContext,
-        uint40 _nextQueueIndex,
+        uint40 _firstSubsequentQueueIndex,
         uint40 _queueLength,
         iOVM_ChainStorageContainer _queueRef
     )
         internal
         view
     {
         // If the queue is not now empty, check the mononoticity of whatever the next batch that will come in is.
-        if (_queueLength - _nextQueueIndex > 0 && _finalContext.numSubsequentQueueTransactions == 0) {
-            _validateContextBeforeEnqueue(
-                _finalContext,
-                _nextQueueIndex,
+        if (_queueLength - _firstSubsequentQueueIndex > 0) {
+            Lib_OVMCodec.QueueElement memory firstSubsequentQueueElement = _getQueueElement(
+                _firstSubsequentQueueIndex,
                 _queueRef
             );
+
+            // If the force inclusion period has passed for an enqueued transaction, it MUST be the next chain element.
```

Comment:
Ahh, good catch on this one.  Is it worth pulling out these error strings into constants since there are duplicates now?

Reply:
Unless you feel strongly, my preference is to not. AFAICT it won't save gas except maybe on deployment.
If you do feel strongly, then I think a better approach would be to extract the duplicate checks into a function.